packages/components/src/CodeEditor.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { FC } from "react";
+import AceEditor from "react-ace";
+
+import "ace-builds/src-noconflict/mode-golang";
+import "ace-builds/src-noconflict/theme-github";
+import "ace-builds/src-noconflict/ext-language_tools";
+import "ace-builds/src-noconflict/keybinding-vim";
+
+interface CodeEditorProps {
+  value: string;
+  onChange: (value: string) => void;
+  oneLine: boolean;
+  width?: number;
+}
+
+export let CodeEditor: FC<CodeEditorProps> = (props) => (
+  <AceEditor
+    value={props.value}
+    mode="golang"
+    theme="github"
+    width={props.width ? props.width + "px" : "500px"}
+    height={props.oneLine ? "1.2em" : "500px"}
+    showGutter={false}
+    highlightActiveLine={false}
+    showPrintMargin={false}
+    onChange={props.onChange}
+    name="UNIQUE_ID_OF_DIV"
```

Comment:
Is this line intentional?

Reply:
Yes, why?

Reply:
I suspected it might have been unnecessary. 

If it's ours to name, I'd probable give it something more specific, like, "Squiggle-CodeEditor." Not super important though.

Reply:
Oh yeah true.... I'm actually mildly concerned that it might screw up with multiple editors, I'll look into that as well

Reply:
Good point. I think React might have some decent way of making custom IDs for elements.

Reply:
Lodash has a decent method for unique IDs.
https://stackoverflow.com/questions/29420835/how-to-generate-unique-ids-for-form-labels-in-react

Reply:
Actually, this one might be best.https://stackoverflow.com/a/71681435

Reply:
Fixed!